<?xml version="1.0"?>
<launch>

	<!-- vehicle model and world -->
	<arg name="est" default="ekf2" />
	<arg name="vehicle" default="iris" />
	<arg name="world" default="$(find uav_gazebo)/worlds/ntu_campus.world" />
	<!-- gazebo configs -->
	<arg name="gui" default="true" />
	<arg name="debug" default="false" />
	<arg name="verbose" default="false" />
	<arg name="paused" default="false" />

	<!-- $(find gazebo_ros)/launch/empty_world.launch -->
	<include file="$(find gazebo_ros)/launch/empty_world.launch">
		<arg name="gui" value="$(arg gui)" />
		<arg name="world_name" value="$(arg world)" />
		<arg name="debug" value="$(arg debug)" />
		<arg name="verbose" value="$(arg verbose)" />
		<arg name="paused" value="$(arg paused)" />
	</include>


	<!-- UAV0 -->
	<group ns="uav0">
		<!-- move uav node -->
		<!-- <node pkg="move_uav" type="move_uav_global_node" name="move_uav_global" output="screen">
			<param name="init_pos_latitude" value="25.017" />
			<param name="init_pos_longitude" value="121.5396" />
			<param name="init_pos_altitude" value="30" />
		</node> -->

		<!-- MAVROS and vehicle configs -->
		<arg name="ID" value="0" />
		<arg name="fcu_local_port" default="$(eval 14580 + arg('ID'))" />
		<arg name="fcu_remote_port" default="$(eval 14540 + arg('ID'))" />
		<arg name="fcu_url" default="udp://:$(arg fcu_remote_port)@localhost:$(arg fcu_local_port)" />
		<arg name="x" value="142" />
		<arg name="y" value="223" />
		<arg name="z" value="18.73" />
		<arg name="R" value="0" />
		<arg name="P" value="0" />
		<arg name="Y" value="0" />
		<!-- PX4 SITL and vehicle spawn -->
		<include file="$(find px4)/launch/single_vehicle_spawn.launch">
			<arg name="x" value="$(arg x)" />
			<arg name="y" value="$(arg y)" />
			<arg name="z" value="$(arg z)" />
			<arg name="R" value="$(arg R)" />
			<arg name="P" value="$(arg P)" />
			<arg name="Y" value="$(arg Y)" />
			<arg name="vehicle" value="$(arg vehicle)" />
			<arg name="mavlink_udp_port" value="$(eval 14560 + arg('ID'))" />
			<arg name="mavlink_tcp_port" value="$(eval 4560 + arg('ID'))" />
			<arg name="ID" value="$(arg ID)" />
			<arg name="gst_udp_port" value="$(eval 5600 + arg('ID'))" />
			<arg name="video_uri" value="$(eval 5600 + arg('ID'))" />
			<arg name="mavlink_cam_udp_port" value="$(eval 14530 + arg('ID'))" />
		</include>
		<!-- MAVROS -->
		<include file="$(find mavros)/launch/px4.launch">
			<arg name="fcu_url" value="$(arg fcu_url)" />
			<arg name="gcs_url" value="" />
			<arg name="tgt_system" value="$(eval 1 + arg('ID'))" />
			<arg name="tgt_component" value="1" />
		</include>

		<!-- Geometric Controller -->
		<node pkg="geometric_controller" type="geometric_controller_node" name="geometric_controller" output="screen">
			<remap from="command/bodyrate_command" to="mavros/setpoint_raw/attitude" />
			<param name="ctrl_mode" value="2" />
			<param name="enable_sim" value="true" />
			<param name="enable_gazebo_state" value="true" />
			<param name="max_acc" value="10.0" />
			<param name="init_pos_x" value="0" />
			<param name="init_pos_y" value="0" />
			<param name="init_pos_z" value="30" />
			<!--Params specific for Iris copter-->
			<param name="attctrl_constant" value="0.3" />
			<param name="normalizedthrust_constant" value="0.06" />
			<param name="normalizedthrust_offset" value="0.1" />
			<param name="Kp_x" value="5.0" />
			<param name="Kp_y" value="5.0" />
			<param name="Kp_z" value="5.0" />
			<param name="Kv_x" value="5.0" />
			<param name="Kv_y" value="5.0" />
			<param name="Kv_z" value="5.0" />
			<!--Use ground truth pose from gazebo plugin-->
			<param name="use_groudtruth" value="true" />
		</node>

		<!-- Transform Publisher -->
		<node pkg="tf" type="static_transform_publisher" name="uav_world_publisher" args="$(arg x) $(arg y) $(arg z) $(arg Y) $(arg P) $(arg R) world uav$(arg ID)_world 100" />
	</group>

	<!-- UAV1 -->
	<group ns="uav1">
		<!-- move uav node -->
		<node pkg="move_uav" type="move_uav_global_node" name="move_uav_global" output="screen">
			<param name="init_pos_latitude" value="25.017" />
			<param name="init_pos_longitude" value="121.5396" />
			<param name="init_pos_altitude" value="31" />
		</node>

		<!-- MAVROS and vehicle configs -->
		<arg name="ID" value="1" />
		<arg name="fcu_local_port" default="$(eval 14580 + arg('ID'))" />
		<arg name="fcu_remote_port" default="$(eval 14540 + arg('ID'))" />
		<arg name="fcu_url" default="udp://:$(arg fcu_remote_port)@localhost:$(arg fcu_local_port)" />
		<arg name="x" value="143" />
		<arg name="y" value="223" />
		<arg name="z" value="18.73" />
		<arg name="R" value="0" />
		<arg name="P" value="0" />
		<arg name="Y" value="0" />
		<!-- PX4 SITL and vehicle spawn -->
		<include file="$(find px4)/launch/single_vehicle_spawn.launch">
			<arg name="x" value="$(arg x)" />
			<arg name="y" value="$(arg y)" />
			<arg name="z" value="$(arg z)" />
			<arg name="R" value="$(arg R)" />
			<arg name="P" value="$(arg P)" />
			<arg name="Y" value="$(arg Y)" />
			<arg name="vehicle" value="$(arg vehicle)" />
			<arg name="mavlink_udp_port" value="$(eval 14560 + arg('ID'))" />
			<arg name="mavlink_tcp_port" value="$(eval 4560 + arg('ID'))" />
			<arg name="ID" value="$(arg ID)" />
			<arg name="gst_udp_port" value="$(eval 5600 + arg('ID'))" />
			<arg name="video_uri" value="$(eval 5600 + arg('ID'))" />
			<arg name="mavlink_cam_udp_port" value="$(eval 14530 + arg('ID'))" />
		</include>
		<!-- MAVROS -->
		<include file="$(find mavros)/launch/px4.launch">
			<arg name="fcu_url" value="$(arg fcu_url)" />
			<arg name="gcs_url" value="" />
			<arg name="tgt_system" value="$(eval 1 + arg('ID'))" />
			<arg name="tgt_component" value="1" />
		</include>

		<!-- Transform Publisher -->
		<node pkg="tf" type="static_transform_publisher" name="uav_world_publisher" args="$(arg x) $(arg y) $(arg z) $(arg Y) $(arg P) $(arg R) world uav$(arg ID)_world 100" />
	</group>

	<!-- UAV2 -->
	<group ns="uav2">
		<!-- move uav node -->
		<node pkg="move_uav" type="move_uav_global_node" name="move_uav_global" output="screen">
			<param name="init_pos_latitude" value="25.017" />
			<param name="init_pos_longitude" value="121.5396" />
			<param name="init_pos_altitude" value="32" />
		</node>

		<!-- MAVROS and vehicle configs -->
		<arg name="ID" value="2" />
		<arg name="fcu_local_port" default="$(eval 14580 + arg('ID'))" />
		<arg name="fcu_remote_port" default="$(eval 14540 + arg('ID'))" />
		<arg name="fcu_url" default="udp://:$(arg fcu_remote_port)@localhost:$(arg fcu_local_port)" />
		<arg name="x" value="144" />
		<arg name="y" value="223" />
		<arg name="z" value="18.73" />
		<arg name="R" value="0" />
		<arg name="P" value="0" />
		<arg name="Y" value="0" />
		<!-- PX4 SITL and vehicle spawn -->
		<include file="$(find px4)/launch/single_vehicle_spawn.launch">
			<arg name="x" value="$(arg x)" />
			<arg name="y" value="$(arg y)" />
			<arg name="z" value="$(arg z)" />
			<arg name="R" value="$(arg R)" />
			<arg name="P" value="$(arg P)" />
			<arg name="Y" value="$(arg Y)" />
			<arg name="vehicle" value="$(arg vehicle)" />
			<arg name="mavlink_udp_port" value="$(eval 14560 + arg('ID'))" />
			<arg name="mavlink_tcp_port" value="$(eval 4560 + arg('ID'))" />
			<arg name="ID" value="$(arg ID)" />
			<arg name="gst_udp_port" value="$(eval 5600 + arg('ID'))" />
			<arg name="video_uri" value="$(eval 5600 + arg('ID'))" />
			<arg name="mavlink_cam_udp_port" value="$(eval 14530 + arg('ID'))" />
		</include>
		<!-- MAVROS -->
		<include file="$(find mavros)/launch/px4.launch">
			<arg name="fcu_url" value="$(arg fcu_url)" />
			<arg name="gcs_url" value="" />
			<arg name="tgt_system" value="$(eval 1 + arg('ID'))" />
			<arg name="tgt_component" value="1" />
		</include>

		<!-- Transform Publisher -->
		<node pkg="tf" type="static_transform_publisher" name="uav_world_publisher" args="$(arg x) $(arg y) $(arg z) $(arg Y) $(arg P) $(arg R) world uav$(arg ID)_world 100" />
	</group>

	<!-- UAV3 -->
	<group ns="uav3">
		<!-- move uav node -->
		<node pkg="move_uav" type="move_uav_global_node" name="move_uav_global" output="screen">
			<param name="init_pos_latitude" value="25.017" />
			<param name="init_pos_longitude" value="121.5396" />
			<param name="init_pos_altitude" value="33" />
		</node>

		<!-- MAVROS and vehicle configs -->
		<arg name="ID" value="3" />
		<arg name="fcu_local_port" default="$(eval 14580 + arg('ID'))" />
		<arg name="fcu_remote_port" default="$(eval 14540 + arg('ID'))" />
		<arg name="fcu_url" default="udp://:$(arg fcu_remote_port)@localhost:$(arg fcu_local_port)" />
		<arg name="x" value="145" />
		<arg name="y" value="223" />
		<arg name="z" value="18.73" />
		<arg name="R" value="0" />
		<arg name="P" value="0" />
		<arg name="Y" value="0" />
		<!-- PX4 SITL and vehicle spawn -->
		<include file="$(find px4)/launch/single_vehicle_spawn.launch">
			<arg name="x" value="$(arg x)" />
			<arg name="y" value="$(arg y)" />
			<arg name="z" value="$(arg z)" />
			<arg name="R" value="$(arg R)" />
			<arg name="P" value="$(arg P)" />
			<arg name="Y" value="$(arg Y)" />
			<arg name="vehicle" value="$(arg vehicle)" />
			<arg name="mavlink_udp_port" value="$(eval 14560 + arg('ID'))" />
			<arg name="mavlink_tcp_port" value="$(eval 4560 + arg('ID'))" />
			<arg name="ID" value="$(arg ID)" />
			<arg name="gst_udp_port" value="$(eval 5600 + arg('ID'))" />
			<arg name="video_uri" value="$(eval 5600 + arg('ID'))" />
			<arg name="mavlink_cam_udp_port" value="$(eval 14530 + arg('ID'))" />
		</include>
		<!-- MAVROS -->
		<include file="$(find mavros)/launch/px4.launch">
			<arg name="fcu_url" value="$(arg fcu_url)" />
			<arg name="gcs_url" value="" />
			<arg name="tgt_system" value="$(eval 1 + arg('ID'))" />
			<arg name="tgt_component" value="1" />
		</include>


		<!-- Transform Publisher -->
		<node pkg="tf" type="static_transform_publisher" name="uav_world_publisher" args="$(arg x) $(arg y) $(arg z) $(arg Y) $(arg P) $(arg R) world uav$(arg ID)_world 100" />
	</group>

	<!-- UAV4 -->
	<group ns="uav4">
		<!-- move uav node -->
		<node pkg="move_uav" type="move_uav_global_node" name="move_uav_global" output="screen">
			<param name="init_pos_latitude" value="25.017" />
			<param name="init_pos_longitude" value="121.5396" />
			<param name="init_pos_altitude" value="34" />
		</node>

		<!-- MAVROS and vehicle configs -->
		<arg name="ID" value="4" />
		<arg name="fcu_local_port" default="$(eval 14580 + arg('ID'))" />
		<arg name="fcu_remote_port" default="$(eval 14540 + arg('ID'))" />
		<arg name="fcu_url" default="udp://:$(arg fcu_remote_port)@localhost:$(arg fcu_local_port)" />
		<arg name="x" value="146" />
		<arg name="y" value="223" />
		<arg name="z" value="18.73" />
		<arg name="R" value="0" />
		<arg name="P" value="0" />
		<arg name="Y" value="0" />
		<!-- PX4 SITL and vehicle spawn -->
		<include file="$(find px4)/launch/single_vehicle_spawn.launch">
			<arg name="x" value="$(arg x)" />
			<arg name="y" value="$(arg y)" />
			<arg name="z" value="$(arg z)" />
			<arg name="R" value="$(arg R)" />
			<arg name="P" value="$(arg P)" />
			<arg name="Y" value="$(arg Y)" />
			<arg name="vehicle" value="$(arg vehicle)" />
			<arg name="mavlink_udp_port" value="$(eval 14560 + arg('ID'))" />
			<arg name="mavlink_tcp_port" value="$(eval 4560 + arg('ID'))" />
			<arg name="ID" value="$(arg ID)" />
			<arg name="gst_udp_port" value="$(eval 5600 + arg('ID'))" />
			<arg name="video_uri" value="$(eval 5600 + arg('ID'))" />
			<arg name="mavlink_cam_udp_port" value="$(eval 14530 + arg('ID'))" />
		</include>
		<!-- MAVROS -->
		<include file="$(find mavros)/launch/px4.launch">
			<arg name="fcu_url" value="$(arg fcu_url)" />
			<arg name="gcs_url" value="" />
			<arg name="tgt_system" value="$(eval 1 + arg('ID'))" />
			<arg name="tgt_component" value="1" />
		</include>


		<!-- Transform Publisher -->
		<node pkg="tf" type="static_transform_publisher" name="uav_world_publisher" args="$(arg x) $(arg y) $(arg z) $(arg Y) $(arg P) $(arg R) world uav$(arg ID)_world 100" />
	</group>

	<!-- UAV5 -->
	<group ns="uav5">
		<!-- move uav node -->
		<node pkg="move_uav" type="move_uav_global_node" name="move_uav_global" output="screen">
			<param name="init_pos_latitude" value="25.017" />
			<param name="init_pos_longitude" value="121.5396" />
			<param name="init_pos_altitude" value="35" />
		</node>

		<!-- MAVROS and vehicle configs -->
		<arg name="ID" value="5" />
		<arg name="fcu_local_port" default="$(eval 14580 + arg('ID'))" />
		<arg name="fcu_remote_port" default="$(eval 14540 + arg('ID'))" />
		<arg name="fcu_url" default="udp://:$(arg fcu_remote_port)@localhost:$(arg fcu_local_port)" />
		<arg name="x" value="147" />
		<arg name="y" value="223" />
		<arg name="z" value="18.73" />
		<arg name="R" value="0" />
		<arg name="P" value="0" />
		<arg name="Y" value="0" />
		<!-- PX4 SITL and vehicle spawn -->
		<include file="$(find px4)/launch/single_vehicle_spawn.launch">
			<arg name="x" value="$(arg x)" />
			<arg name="y" value="$(arg y)" />
			<arg name="z" value="$(arg z)" />
			<arg name="R" value="$(arg R)" />
			<arg name="P" value="$(arg P)" />
			<arg name="Y" value="$(arg Y)" />
			<arg name="vehicle" value="$(arg vehicle)" />
			<arg name="mavlink_udp_port" value="$(eval 14560 + arg('ID'))" />
			<arg name="mavlink_tcp_port" value="$(eval 4560 + arg('ID'))" />
			<arg name="ID" value="$(arg ID)" />
			<arg name="gst_udp_port" value="$(eval 5600 + arg('ID'))" />
			<arg name="video_uri" value="$(eval 5600 + arg('ID'))" />
			<arg name="mavlink_cam_udp_port" value="$(eval 14530 + arg('ID'))" />
		</include>
		<!-- MAVROS -->
		<include file="$(find mavros)/launch/px4.launch">
			<arg name="fcu_url" value="$(arg fcu_url)" />
			<arg name="gcs_url" value="" />
			<arg name="tgt_system" value="$(eval 1 + arg('ID'))" />
			<arg name="tgt_component" value="1" />
		</include>


		<!-- Transform Publisher -->
		<node pkg="tf" type="static_transform_publisher" name="uav_world_publisher" args="$(arg x) $(arg y) $(arg z) $(arg Y) $(arg P) $(arg R) world uav$(arg ID)_world 100" />
	</group>

	<!-- UAV6 -->
	<group ns="uav6">
		<!-- move uav node -->
		<node pkg="move_uav" type="move_uav_global_node" name="move_uav_global" output="screen">
			<param name="init_pos_latitude" value="25.017" />
			<param name="init_pos_longitude" value="121.5396" />
			<param name="init_pos_altitude" value="36" />
		</node>

		<!-- MAVROS and vehicle configs -->
		<arg name="ID" value="6" />
		<arg name="fcu_local_port" default="$(eval 14580 + arg('ID'))" />
		<arg name="fcu_remote_port" default="$(eval 14540 + arg('ID'))" />
		<arg name="fcu_url" default="udp://:$(arg fcu_remote_port)@localhost:$(arg fcu_local_port)" />
		<arg name="x" value="148" />
		<arg name="y" value="223" />
		<arg name="z" value="18.73" />
		<arg name="R" value="0" />
		<arg name="P" value="0" />
		<arg name="Y" value="0" />
		<!-- PX4 SITL and vehicle spawn -->
		<include file="$(find px4)/launch/single_vehicle_spawn.launch">
			<arg name="x" value="$(arg x)" />
			<arg name="y" value="$(arg y)" />
			<arg name="z" value="$(arg z)" />
			<arg name="R" value="$(arg R)" />
			<arg name="P" value="$(arg P)" />
			<arg name="Y" value="$(arg Y)" />
			<arg name="vehicle" value="$(arg vehicle)" />
			<arg name="mavlink_udp_port" value="$(eval 14560 + arg('ID'))" />
			<arg name="mavlink_tcp_port" value="$(eval 4560 + arg('ID'))" />
			<arg name="ID" value="$(arg ID)" />
			<arg name="gst_udp_port" value="$(eval 5600 + arg('ID'))" />
			<arg name="video_uri" value="$(eval 5600 + arg('ID'))" />
			<arg name="mavlink_cam_udp_port" value="$(eval 14530 + arg('ID'))" />
		</include>
		<!-- MAVROS -->
		<include file="$(find mavros)/launch/px4.launch">
			<arg name="fcu_url" value="$(arg fcu_url)" />
			<arg name="gcs_url" value="" />
			<arg name="tgt_system" value="$(eval 1 + arg('ID'))" />
			<arg name="tgt_component" value="1" />
		</include>


		<!-- Transform Publisher -->
		<node pkg="tf" type="static_transform_publisher" name="uav_world_publisher" args="$(arg x) $(arg y) $(arg z) $(arg Y) $(arg P) $(arg R) world uav$(arg ID)_world 100" />
	</group>

	<!-- UAV7 -->
	<group ns="uav7">
		<!-- move uav node -->
		<node pkg="move_uav" type="move_uav_global_node" name="move_uav_global" output="screen">
			<param name="init_pos_latitude" value="25.017" />
			<param name="init_pos_longitude" value="121.5396" />
			<param name="init_pos_altitude" value="37" />
		</node>

		<!-- MAVROS and vehicle configs -->
		<arg name="ID" value="7" />
		<arg name="fcu_local_port" default="$(eval 14580 + arg('ID'))" />
		<arg name="fcu_remote_port" default="$(eval 14540 + arg('ID'))" />
		<arg name="fcu_url" default="udp://:$(arg fcu_remote_port)@localhost:$(arg fcu_local_port)" />
		<arg name="x" value="149" />
		<arg name="y" value="223" />
		<arg name="z" value="18.73" />
		<arg name="R" value="0" />
		<arg name="P" value="0" />
		<arg name="Y" value="0" />
		<!-- PX4 SITL and vehicle spawn -->
		<include file="$(find px4)/launch/single_vehicle_spawn.launch">
			<arg name="x" value="$(arg x)" />
			<arg name="y" value="$(arg y)" />
			<arg name="z" value="$(arg z)" />
			<arg name="R" value="$(arg R)" />
			<arg name="P" value="$(arg P)" />
			<arg name="Y" value="$(arg Y)" />
			<arg name="vehicle" value="$(arg vehicle)" />
			<arg name="mavlink_udp_port" value="$(eval 14560 + arg('ID'))" />
			<arg name="mavlink_tcp_port" value="$(eval 4560 + arg('ID'))" />
			<arg name="ID" value="$(arg ID)" />
			<arg name="gst_udp_port" value="$(eval 5600 + arg('ID'))" />
			<arg name="video_uri" value="$(eval 5600 + arg('ID'))" />
			<arg name="mavlink_cam_udp_port" value="$(eval 14530 + arg('ID'))" />
		</include>
		<!-- MAVROS -->
		<include file="$(find mavros)/launch/px4.launch">
			<arg name="fcu_url" value="$(arg fcu_url)" />
			<arg name="gcs_url" value="" />
			<arg name="tgt_system" value="$(eval 1 + arg('ID'))" />
			<arg name="tgt_component" value="1" />
		</include>


		<!-- Transform Publisher -->
		<node pkg="tf" type="static_transform_publisher" name="uav_world_publisher" args="$(arg x) $(arg y) $(arg z) $(arg Y) $(arg P) $(arg R) world uav$(arg ID)_world 100" />
	</group>

	<!-- UAV8 -->
	<group ns="uav8">
		<!-- move uav node -->
		<node pkg="move_uav" type="move_uav_global_node" name="move_uav_global" output="screen">
			<param name="init_pos_latitude" value="25.017" />
			<param name="init_pos_longitude" value="121.5396" />
			<param name="init_pos_altitude" value="38" />
		</node>

		<!-- MAVROS and vehicle configs -->
		<arg name="ID" value="8" />
		<arg name="fcu_local_port" default="$(eval 14580 + arg('ID'))" />
		<arg name="fcu_remote_port" default="$(eval 14540 + arg('ID'))" />
		<arg name="fcu_url" default="udp://:$(arg fcu_remote_port)@localhost:$(arg fcu_local_port)" />
		<arg name="x" value="149" />
		<arg name="y" value="224" />
		<arg name="z" value="18.73" />
		<arg name="R" value="0" />
		<arg name="P" value="0" />
		<arg name="Y" value="0" />
		<!-- PX4 SITL and vehicle spawn -->
		<include file="$(find px4)/launch/single_vehicle_spawn.launch">
			<arg name="x" value="$(arg x)" />
			<arg name="y" value="$(arg y)" />
			<arg name="z" value="$(arg z)" />
			<arg name="R" value="$(arg R)" />
			<arg name="P" value="$(arg P)" />
			<arg name="Y" value="$(arg Y)" />
			<arg name="vehicle" value="$(arg vehicle)" />
			<arg name="mavlink_udp_port" value="$(eval 14560 + arg('ID'))" />
			<arg name="mavlink_tcp_port" value="$(eval 4560 + arg('ID'))" />
			<arg name="ID" value="$(arg ID)" />
			<arg name="gst_udp_port" value="$(eval 5600 + arg('ID'))" />
			<arg name="video_uri" value="$(eval 5600 + arg('ID'))" />
			<arg name="mavlink_cam_udp_port" value="$(eval 14530 + arg('ID'))" />
		</include>
		<!-- MAVROS -->
		<include file="$(find mavros)/launch/px4.launch">
			<arg name="fcu_url" value="$(arg fcu_url)" />
			<arg name="gcs_url" value="" />
			<arg name="tgt_system" value="$(eval 1 + arg('ID'))" />
			<arg name="tgt_component" value="1" />
		</include>

		<!-- Transform Publisher -->
		<node pkg="tf" type="static_transform_publisher" name="uav_world_publisher" args="$(arg x) $(arg y) $(arg z) $(arg Y) $(arg P) $(arg R) world uav$(arg ID)_world 100" />
	</group>

	<!-- UAV9 -->
	<group ns="uav9">
		<!-- move uav node -->
		<node pkg="move_uav" type="move_uav_global_node" name="move_uav_global" output="screen">
			<param name="init_pos_latitude" value="25.017" />
			<param name="init_pos_longitude" value="121.5396" />
			<param name="init_pos_altitude" value="39" />
		</node>

		<!-- MAVROS and vehicle configs -->
		<arg name="ID" value="9" />
		<arg name="fcu_local_port" default="$(eval 14580 + arg('ID'))" />
		<arg name="fcu_remote_port" default="$(eval 14540 + arg('ID'))" />
		<arg name="fcu_url" default="udp://:$(arg fcu_remote_port)@localhost:$(arg fcu_local_port)" />
		<arg name="x" value="149" />
		<arg name="y" value="225" />
		<arg name="z" value="18.73" />
		<arg name="R" value="0" />
		<arg name="P" value="0" />
		<arg name="Y" value="0" />
		<!-- PX4 SITL and vehicle spawn -->
		<include file="$(find px4)/launch/single_vehicle_spawn.launch">
			<arg name="x" value="$(arg x)" />
			<arg name="y" value="$(arg y)" />
			<arg name="z" value="$(arg z)" />
			<arg name="R" value="$(arg R)" />
			<arg name="P" value="$(arg P)" />
			<arg name="Y" value="$(arg Y)" />
			<arg name="vehicle" value="$(arg vehicle)" />
			<arg name="mavlink_udp_port" value="$(eval 14560 + arg('ID'))" />
			<arg name="mavlink_tcp_port" value="$(eval 4560 + arg('ID'))" />
			<arg name="ID" value="$(arg ID)" />
			<arg name="gst_udp_port" value="$(eval 5600 + arg('ID'))" />
			<arg name="video_uri" value="$(eval 5600 + arg('ID'))" />
			<arg name="mavlink_cam_udp_port" value="$(eval 14530 + arg('ID'))" />
		</include>
		<!-- MAVROS -->
		<include file="$(find mavros)/launch/px4.launch">
			<arg name="fcu_url" value="$(arg fcu_url)" />
			<arg name="gcs_url" value="" />
			<arg name="tgt_system" value="$(eval 1 + arg('ID'))" />
			<arg name="tgt_component" value="1" />
		</include>

		<!-- Transform Publisher -->
		<node pkg="tf" type="static_transform_publisher" name="uav_world_publisher" args="$(arg x) $(arg y) $(arg z) $(arg Y) $(arg P) $(arg R) world uav$(arg ID)_world 100" />
	</group>

</launch>